### PR TITLE
refactor: remove deprecated GeneratedVaadinDialog

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Removed deprecated `GeneratedVaadinDialog` class and moved a few methods to `Dialog`.

## Type of change

- Breaking change

### CODE REVIEW

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 1]
```diff
@@ -917,6 +919,45 @@ protected void onAttach(AttachEvent attachEvent) {
         initHeaderFooterRenderer();
     }
 
+    /**
+     * <p>
+     * Description copied from corresponding location in WebComponent:
```
Comment (sissbruecker) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:924]: Let's clean up the generated stuff that is not helpful for Flow users like `Description copied from corresponding location in WebComponent:`

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 2]
```diff
@@ -917,6 +919,45 @@ protected void onAttach(AttachEvent attachEvent) {
...
+     * default) means that the {@code aria-label} attribute is not present at
+     * all.
+     * <p>
+     * This property is not synchronized automatically from the client side, so
+     * the returned value may not be the same as in client side.
+     * </p>
+     *
+     * @return the {@code ariaLabel} property from the webcomponent
+     */
+    protected String getAriaLabelString() {
```
Comment (sissbruecker) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:938]:
```suggestion
    protected String getAriaLabel() {
```

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 3]
```diff
@@ -179,6 +188,22 @@ public String getHeight() {
         }
     }
 
+    /**
+     * `opened-changed` event is sent when the overlay opened state changes.
```
Comment (tomivirkki) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:192]: Backtick notation doesn't work correctly in JavaDoc. See a [similar comment in context-menu base](https://github.com/vaadin/flow-components/blob/master/vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java#L313).

Screenshot 2022-12-09 at 12 33 34

**Review by sissbruecker — APPROVED** [review]

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java (hunk 3 above)
Comment (web-padawan) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java]: Done.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**